src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -475,6 +595,22 @@ def load_sub_model(
         # else load from the root directory
         loaded_sub_model = load_method(cached_folder, **loading_kwargs)
 
+    if isinstance(loaded_sub_model, torch.nn.Module) and isinstance(device_map, dict):
+        # remove hooks
+        remove_hook_from_module(loaded_sub_model, recurse=True)
+        needs_offloading_to_cpu = device_map[""] == "cpu"
+
+        if needs_offloading_to_cpu:
```

Comment:
why do we need to dispatch it again that we already passed `device_map` to the `load_method`?

Reply:
* `transformers` doesn't do `force_hooks` in its loading. It's needed as explained above. 
* For the "balanced" strategy we're implementing here, we need to have the `main_device` set to 0. This is not the case in any of the loading methods we rely on currently. Hence the need.

Reply:
Ohh, ok, I understand now. Thanks for explaining! 
do we still prefer to have this supported from `Transformers`? seems like they're open to this, no?

Reply:
Sure, I will ping Arthur after we land the PR so as to not block this PR. Hopefully, that's okay?

Reply:
yeah sure we can explore that after merging this PR